pkg/virt-controller/services/template.go

```diff
@@ -94,6 +96,17 @@ func (t *templateService) RenderLaunchManifest(vm *v1.VirtualMachine) (*kubev1.P
 				},
 			})
 		}
+		if volume.RegistryDisk != nil && volume.RegistryDisk.ImagePullSecret != "" {
+			imagePullSecrets = append(imagePullSecrets, kubev1.LocalObjectReference{
+				Name: volume.RegistryDisk.ImagePullSecret,
+			})
+		}
+	}
+
+	if t.imagePullSecret != "" {
+		imagePullSecrets = append(imagePullSecrets, kubev1.LocalObjectReference{
+			Name: t.imagePullSecret,
```

Comment:
if two registry disks reference the same imagePullSecret, will duplicate entries in this list be allowed by k8s?

Reply:
I can deduplicate the list. I know it doesn't cause a problem to have it listed twice.  We could also put it higher in the tree, not a per-disk property. I went back and forth on this.